plugins/newsdownloader.koplugin/main.lua

```diff
@@ -48,415 +58,648 @@ end
+	 keep_menu_open = true,
+	 callback = function() self:removeNewsButKeepFeedConfig() end,
+      },
+      {
+	 text = _("Settings"),
+	 sub_item_table = {
+	    {
+	       text = _("Edit feeds configuration file"),
+	       keep_menu_open = true,
+	       callback = function() self:changeFeedConfig() end,
+	    },
+	    {
+	       text = _("Set custom download folder"),
+	       keep_menu_open = true,
+	       callback = function() self:setCustomDownloadDirectory() end,
+	    },
+	    {
+	       text = _("Never download images"),
+	       keep_menu_open = true,
+	       checked_func = function()
+		  return self.settings:isTrue("never_download_images")
+	       end,
+	       callback = function()
+		  self.settings:toggle("never_download_images")
+		  self.settings:flush()
+	       end,
+	    },
+	 },
+      },
+      {
+	 text = _("Help"),
+	 keep_menu_open = true,
+	 callback = function()
+	    UIManager:show(InfoMessage:new{
+			      text = T(_("News downloader retrieves RSS and Atom news entries and stores them to:\n%1\n\nEach entry is a separate html file, that can be browsed by KOReader file manager.\nItems download limit can be configured in Settings."),
+				       BD.dirpath(self.download_dir))
+	    })
+	 end,
+      },
+   }
+   return sub_item_table
+end
+-- lazyInitialization sets up variables that point to the
+-- downloads folder and the feeds configuration file
 function NewsDownloader:lazyInitialization()
-   if not initialized then
-        logger.dbg("NewsDownloader: obtaining news folder")
-        news_downloader_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), news_downloader_config_file))
-        if news_downloader_settings:has(config_key_custom_dl_dir) then
-            news_download_dir_path = news_downloader_settings:readSetting(config_key_custom_dl_dir)
-        else
-            news_download_dir_path = ("%s/%s/"):format(DataStorage:getFullDataDir(), news_download_dir_name)
-        end
-
-        if not lfs.attributes(news_download_dir_path, "mode") then
-            logger.dbg("NewsDownloader: Creating initial directory")
-            lfs.mkdir(news_download_dir_path)
-        end
-        feed_config_path = news_download_dir_path .. feed_config_file_name
-
-        if not lfs.attributes(feed_config_path, "mode") then
-            logger.dbg("NewsDownloader: Creating initial feed config.")
-            FFIUtil.copyFile(FFIUtil.joinPath(self.path, feed_config_file_name),
-                         feed_config_path)
-        end
-        initialized = true
-    end
+   if not self.initialized then
+      logger.dbg("NewsDownloader: obtaining news folder")
+      self.settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), self.news_config_file))
+      -- Check to see if a custom download directory has been set
+      if self.settings:has(self.config_key_custom_dl_dir) then
+	 self.download_dir = self.settings:readSetting(self.config_key_custom_dl_dir)
```

Comment:
Also indentation mishap here. (Unclear on GitHub, but perhaps a few TABs are intruding?).

(Sidebar: we all hate tabs, but we're living with the existing house style, unfortunately ;p).

Reply:
I'm using Emacs and haven't quite figured out how to customize Lua mode. If anyone else uses Emacs, I'd love to hear how they're configuring their `.emacs`

Reply:
(Othersidebar: I don't love tabs !)
And yes, definitely 4 spaces per level. GitHub's "File changed" diff shows too much stuff you didn't really change - will make it easier to review once the indentation gets back to the original.
No idea about Emacs.

Reply:
How about I delete this PR and try again?

Reply:
You don't need to: you can just edit/fix your files and push another commit to this same PR.